docs/support/tutorials/nextflow-puhti.md

```diff
@@ -0,0 +1,139 @@
+#SBATCH --cpus-per-task=xx # change as you need
+
+export PROJAPPL=/projappl/project_XXX
+module load bioconda
+source activate nextflow
+
+# Actual Nextflow command here
+nextflow run  <workflow.nf> [options]
+```
+
+## 6. (Optional) tutorial - Demonstration of WtP Nextflow pipeline
+
+WtP is a scalable and easy-to-use workflow for phage identification and analysis. More details about the pipeline can be found [here](https://github.com/replikation/What_the_Phage).
+
+Login to Puhti and install Nextflow using conda environment as instructed above.
+
+## Set-up WtP pipeline on Puhti
+
+You can either clone WtP GitHub repository to your project directory in scratch drive  as below:
+
+```
+cd  /Path_on_scratch
+git clone  https://github.com/replikation/What_the_Phage.git
+nextflow run /Path_on_scratch/What_the_Phage/phage.nf --help
+```
+
+or pull the Nextflow pipeline from DockerHub:
+
+```
+nextflow run replikation/What_the_Phage  -r v0.8.0 --help
+```
+
+In either case, pay attention to the versions of different singularity containers as mentioned in containers.config file. 
+
+### Bring your WtP singularity images to Puhti
+
+WtP is a multi-container pipeline requiring as many as 21 singularity images (see [here](https://github.com/replikation/What_the_Phage/blob/master/configs/container.config) for further details) at the time of writing this tutorial. All these containers are downloaded in cPouta environment to avoid any build failures of singularity images on Puhti due to the lack of privileged access for users. For the sake of this tutorial, the downloaded singularity images are uploaded to ``Allas`` which is an object storage environment at CSC.
+
+The images on `Allas` object storage can be downloaded to your project directory on scratch as below:
+```
+mkdir /Path_on_scratch/What_the_Phage/singularity
+cd /Path_on_scratch/What_the_Phage/singularity
+wget https://a3s.fi/puhti_singularity/WtP_singularity.tar.gz
+tar -xavf WtP_singularity.tar.gz
+```
+
+### Run WtP pipelines as a batch job on Puhti:
+
+Submit the following batch script to run nextflow pipeline:
+```
+#!/bin/bash
+#SBATCH --time=01:00:00
+#SBATCH --partition=small
+#SBATCH --account=project_XXX
+#SBATCH --cpus-per-task=4
+
+# Set-up Nextflow on Puhti
+export TMPDIR=$PWD
+export PROJAPPL=/projappl/project_XXX
+module load bioconda
+source activate nextflow
+
+# Nextflow command here
+nextflow run /Path_on_scratch/What_the_Phage/phage.nf --fasta /Path_on_scratch/What_the_Phage/test-data/OX2_draft.fa --cores 4 --output results -profile local,singularity --cachedir /Path_on_scratch//What_the_Phage/singularity --databases /Path_on_scratch/What_the_Phage/databases/WtP_databases --workdir
+/Path_on_scratch/What_the_Phage/workflow-phages-username 
+```
+
+Please note that Singularity images in this example should be in the folder:"/Path_on_scratch//What_the_Phage/singularity". Otherwise, Nextflow tries to download Singularity images again and eventually may prone to failures.
```

Comment:
```suggestion
Please note that Singularity images in this example should be in the folder:"/Path_on_scratch//What_the_Phage/singularity". Otherwise, Nextflow tries to download Singularity images again and eventually may be prone to failures.
```